From 0000000000000000000000000000000000000001 Mon Sep 17 00:00:00 2001
From: Santosh Barnwal <sana@qti.qualcomm.com>
Date: Sun, 23 Aug 2026 11:10:00 +0000
Subject: [PATCH] arm64: dts: qcom: nord: add GIC-v3 interrupt controller

Nord uses the GIC-700 interrupt controller at 0x17000000, matching
the layout of other Oryon-based SoCs. Add the intc node with 18
GICR redistributors (one per Oryon CPU) at stride 0x40000 and the
GIC-ITS block at 0x17040000.

IPCat APSS_GIC700_GICD_APSS: base 0x17000000, size 0x510000
  GICD: 0x17000000 .. 0x17010000 (0x10000)
  ITS:  0x17040000 .. 0x17080000 (0x40000)
  GICR: 0x17080000 .. 0x17500000 (18 * 0x40000 = 0x480000)

Signed-off-by: Santosh Barnwal <sana@qti.qualcomm.com>

Upstream-Status: Pending

---
 arch/arm64/boot/dts/qcom/nord.dtsi | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/nord.dtsi b/arch/arm64/boot/dts/qcom/nord.dtsi
--- a/arch/arm64/boot/dts/qcom/nord.dtsi
+++ b/arch/arm64/boot/dts/qcom/nord.dtsi
@@ -3,6 +3,8 @@
  * Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
  */

+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
 / {
 	interrupt-parent = <&intc>;

@@ -295,5 +297,26 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		ranges;
+
+		intc: interrupt-controller@17000000 {
+			compatible = "arm,gic-v3";
+			reg = <0x0 0x17000000 0x0 0x10000>,
+			      <0x0 0x17080000 0x0 0x480000>;
+			interrupts = <GIC_PPI 9 IRQ_TYPE_LEVEL_HIGH>;
+			#interrupt-cells = <3>;
+			interrupt-controller;
+			#redistributor-regions = <1>;
+			redistributor-stride = <0x0 0x40000>;
+			#address-cells = <2>;
+			#size-cells = <2>;
+			ranges;
+
+			its: msi-controller@17040000 {
+				compatible = "arm,gic-v3-its";
+				reg = <0x0 0x17040000 0x0 0x40000>;
+				msi-controller;
+				#msi-cells = <1>;
+			};
+		};
 	};
 };
